torch/utils/data/datapipes/utils/decoder.py

```diff
@@ -241,16 +262,16 @@ class Decoder:
     handlers until some handler returns something other than None.
     """
 
-    def __init__(self, handlers):
-        self.handlers = handlers
+    def __init__(self, *handler, key_fn=_default_key_fn):
+        self.handlers = list(handler) if handler else []
+        self.key_fn = key_fn
 
-    def add_handler(self, handler):
+    # Insert new handler from the beginning of handlers list to make sure the new
+    # handler having the highest priority
+    def add_handler(self, *handler):
         if not handler:
             return
-        if not self.handlers:
-            self.handlers = [handler]
-        else:
-            self.handlers.append(handler)
+        self.handlers[0: 0] = list(handler)
```

Comment:
That syntax looks quite odd. Since we are not looping here, maybe:

```suggestion
        self.handlers = list(handler) + self.handlers
```